stock_storage_type/models/stock_location.py

```diff
@@ -55,6 +54,81 @@ class StockLocation(models.Model):
         "location_id",
         string="Storage locations sequences",
     )
+    location_is_empty = fields.Boolean(
+        compute="_compute_location_is_empty",
+        store=True,
+        help="technical field: True if the location is empty "
+        "and there is no pending incoming products in the location"
+    )
+
+    in_move_ids = fields.One2many(
+        'stock.move', 'location_dest_id',
+        domain=[('state', 'not in', ('draft', 'done', 'cancel'))],
```

Comment:
Shouldn't we use IN with the other states to be sure the index is used?

Reply:
It could help, but I'm not 100% sure the index will be used.